Python/tigre/utilities/Ax.py

```diff
@@ -18,7 +18,7 @@ def Ax(img, geo, angles, projection_type="Siddon"):
     geox.cast_to_single()
     #geox.checknans()
 
-    if all(img.shape != geox.nVoxel):
+    if abs(img.shape - geox.nVoxel).max()>1e-8:

```

Comment:
```img.shape```  and ```geox.nVoxel``` are both supposed to be integers, so I don't think tolerance is necessary. If you are worried about the case where nVoxel is floating point value, how about checking like https://github.com/CERN/TIGRE/blob/41ffa73421512b49515e749a7d809d8d8c0af948/Python/tigre/utilities/Ax.py#L7 ? I think it is clearer to understand.

Reply:
I think the problem I had is, after applied the conversion of geox to float32, geox.nVoxel becomes float32 as well, so it is not comparable to img.shape (int).... @genusn

Reply:
Hum,  `nVoxel` cast to `float32` at this stage, weird design decision (from my side). Its then later converted to int again, as the CUDA code only allows for ints there 

https://github.com/CERN/TIGRE/blob/41ffa73421512b49515e749a7d809d8d8c0af948/Python/tigre/Source/_types.pxd#L5

In any case, @genusn is correct here. `float32` may nor be able to have infinite precision for decimal numbers, but any _integer_ up to `2^31` should be accurately represented, so the equality applies.

Reply:
@AnderBiguri That is just my quick "fix" to allow me to test other issue. I am sure there are better ways to do it. Feel free to change this.  I am still testing python version algorithms. SIRT seems not working, and for small problems (2D and 3D cases), POCS family seems OK. Problem occurs when we have large cases. However, MLEM works well in any case.

Reply:
Sure @yliu88au , I am happy to fix all the notes I am leaving in this code if you want. I still will leave them, otherwise I will forget :)

Reply:
@yliu88au Thank you for explaining the reason. I didn't see geox.cast_to_single(). I propose moving this shape check block to the head of the function Ax.

Reply:
@genusn I agree, maybe we should check the dimension of img at the beginning using geo.nVoxel, instead of geox.nVoxel (which is converted to float32).